### PR TITLE
BLEU metric card

### PR DESCRIPTION
**Pull request by emibaylor**
Add BLEU metric card

### CODE REVIEW

**Comment by sashavor** [comment]
Some thoughts:
- For values, e.g. "Defaults to False", I would put False in code: `False`. Same for : "Defaults to `4`."
- I would put the following remark in "Limitations": 
> "BLEU's output is always a number between 0 and 1. This value indicates how similar the candidate text is to the reference texts, with values closer to 1 representing more similar texts. Few human translations will attain a score of 1, since this would indicate that the candidate is identical to one of the reference translations. For this reason, it is not necessary to attain a score of 1. Because there are more opportunities to match, adding additional reference translations will increase the BLEU score."

- Add some values from the original BLEU paper (https://aclanthology.org/P02-1040.pdf)

**Review by lhoestq — COMMENTED** [review]
Cool thank you ! Just one comment about the syntax of the python code snippets:

File: metrics/bleu/README.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    ["foo", "bar" "foobar"]
+]
+references = [
+    [["hello", "there", "general", "kenobi"]],
+    [["foo", "bar", "foobar"]]
+]
+bleu = datasets.load_metric("bleu")
+results = bleu.compute(predictions=predictions, references=references)
+print(results)
+>>> {'bleu': 0.6370964381207871, 'precisions': [0.8333333333333334, 0.75, 1.0, 1.0], 'brevity_penalty': 0.7165313105737893, 'length_ratio': 0.75, 'translation_length': 6, 'reference_length': 8}
```
[the remark this answers is not in the document]
Comment (emibaylor) [reply metrics/bleu/README.md:28]: Whoops! Somehow I completely missed this! That makes total sense though, I'll fix it here and will write the examples in doctest style from here on, with the plan to go back and fix the previous cards afterwards. Thanks!

File: metrics/bleu/README.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+...     ["hello", "there", "general", "kenobi",
+...     ["foo", "bar" "foobar"]
+... ]
+>>> references = [
+...     [["hello", "there", "general", "kenobi"]],
+...     [["foo", "bar", "foobar"]]
+... ]
+>>> bleu = datasets.load_metric("bleu")
+>>> results = bleu.compute(predictions=predictions, references=references)
+... print(results)
```
Comment (lhoestq) [inline metrics/bleu/README.md:27]:
```suggestion
>>> print(results)
```

File: metrics/bleu/README.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+...     ["hello", "there", "general", "kenobi",
+...     ["foo", "bar" "foobar"]
+... ]
+>>> references = [
+...     [["hello", "there", "general", "kenobi"], ["hello", "there", "!"]],
+...     [["foo", "bar", "foobar"]]
+... ]
+>>> bleu = datasets.load_metric("bleu")
+>>> results = bleu.compute(predictions=predictions, references=references)
+... print(results)
```
Comment (lhoestq) [inline metrics/bleu/README.md:87]:
```suggestion
>>> print(results)
```

**Review by lhoestq — APPROVED** [review]
Thank you :)

Feel free to apply the suggestions and merge:

**Pull request merged**